src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -105,6 +114,20 @@ public override string SubProtocol
                 websocketControl.SupportedProtocols.Add(subProtocol);
             }
 
+            if (MessageWebSocketClientCertificateSupported)
```

Comment:
In the new iteration, I'm throwing a NotSupportedException if an app has specified *some* client certs and the necessary WinRT API surface isn't available. The exception is being thrown as part of _WinRTWebSocket.ConnectAsync_ for three reasons:

1. Throwing while setting the ClientWebSocketOptions would require a large amount of changes in the PAL, since we should only throw if a client cert is actually added to the ClientWebSocketOptions.ClientCertificates collection. At a minimum, the generic X509CertificateCollection class' Add(...) method would need to be extended for that to happen (not sure if that's even feasible) and the necessary information would need to be piped through the ClientWebSocketOptions and WebSocketHandle classes.
2. _WebSocketHandle.ConnectAsyncCore_ (see [WebSocketHandle.WinRT.cs](https://github.com/dotnet/corefx/blob/master/src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.WinRT.cs)) calls into _WinRTWebSocket.ConnectAsync_ (this method) and catches all exceptions. Those exceptions are then used as the inner exception for a new WebSocketException. Thus, app developers would not need to add a new "catch" statement just to handle this scenario (they should already be catching WebSocketException).
3. A descriptive exception message (which I'm adding) still allows a developer to figure out what went wrong.